### PR TITLE
Remove codepoint locations struct

### PR DESCRIPTION
**Pull request by dandclark**
The "codepoint locations" struct is defined but not really referenced throughout the rest of the spec. Its intention was to allow for multiple ranges of cached codepoint rects, but the getters and setters for character bounds spec assume that only one such range exists.

Since it doesn't seem that support for multiple ranges ended up being needed, remove the "codepoint locations" concept so that the setters/getters are consistent with the actual shape of EditContext's data structures.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -124,16 +125,9 @@ <h4>EditContext state</h4>
...
-            <ul>
-                <li><dfn>start index</dfn> which is an offset into [=text=] that respresents the position before the first codepoint whose location is reported by the first member of [=codepoint rects=] array.</li>
-                <li><dfn>codepoint rects</dfn> is an array of {{DOMRect}} defining the bounding box of each codepoint.</li>
-            </ul>
             <p class="note">
-                [=Codepoint locations=] provides the means for the user agent to query multiple ranges of [=text=] for positioning information.
-                Different platforms may require different positions to be cached to fulfill queries from the [=Text Input Service=] and this API shape allows for that flexibility.
-                It would be good if the API could be simplified to require just the positions of the actively composed text along with the [=selection bounds=] and [=control bounds=].
-                Exploration is needed to see if that is possible.
+              [=Codepoint rects=] provides the means for the user agent to query a range of [=text=] for positioning information.
```
Comment (snianu) [inline index.html]: Do we need to describe what `positioning information.` means? (Can be in a separate PR if you think that information would be useful).

File: index.html [hunk 2]
```diff
@@ -124,16 +125,9 @@ <h4>EditContext state</h4>
...
+              Different platforms may require different positions to be cached to fulfill queries from the [=Text Input Service=].
```
Comment (snianu) [inline index.html]: >  Different platforms may require different positions to be cached to fulfill queries from the [=Text Input Service=].

Perhaps a note with some examples would be helpful to understand the differences in how platforms use these positions? (Also, can be in a separate PR).

**Review by snianu — APPROVED** [review]

File: index.html (hunk 1 above)
Comment (dandclark) [reply index.html]: Elaborated a bit on how it's used. In a follow up PR I'll give more info on the client coordinate system that's used.

**Review by snianu — APPROVED** [review]

File: index.html (hunk 2 above)
Comment (dandclark) [reply index.html]: I didn't really have anything specific in mind here, this was meant to express that `CharacterBoundsUpdateEvent` may be fired differently by different UAs. Updated this to mention `CharacterBoundsUpdateEvent`
